### PR TITLE
ACM-PCA: Use serialisable types instead of native objs

### PR DESCRIPTION
**Pull request by viren-nadkarni**
This is a follow up to https://github.com/getmoto/moto/pull/7958

The `ImportCertificateAuthorityCertificate` would create a `cryptography.hazmat.bindings._rust.x509.Certificate` object as assign it as a `CertificateAuthority` class member. This prevents it from being serialised with pickle.

This PR reworks the certificate and private key attributes so that all members are picklable.

### CODE REVIEW

**Comment by bblommers** [comment]
Hey @viren-nadkarni! Do you want to add some tests for this as well in Moto? Just to prevent it from breaking again in the future

File: moto/acmpca/models.py [hunk 1]
```diff
@@ -287,7 +297,7 @@ def not_valid_before(self) -> Optional[float]:
     def import_certificate_authority_certificate(
         self, certificate: bytes, certificate_chain: Optional[bytes]
     ) -> None:
-        self.certificate = x509.load_pem_x509_certificate(certificate)
+        self.certificate_bytes = certificate
```
Comment (bblommers) [inline moto/acmpca/models.py:300]:
```suggestion
        x509.load_pem_x509_certificate(certificate)
        self.certificate_bytes = certificate
```
Ideally we keep the validation that `certificate`-argument contains a valid X509 certificate. 

As an alternative, something like `assert self.certificate is not None` would also work, as that loads the certificate behind the scenes, but it's less obvious why we're making that assertion.

**Review by bblommers — CHANGES_REQUESTED** [review]

File: moto/acmpca/models.py (hunk 1 above)
Comment (viren-nadkarni) [reply moto/acmpca/models.py:306]: Great catch, added the check and test in ecfa1b336. Thanks @bblommers !

**Review by bblommers — APPROVED** [review]
Great, thanks @viren-nadkarni!

**Pull request merged**